### ISSUE TITLE
Autosave does not automatically save when connection returns

### ISSUE THREAD

**Issue by lin-d-hop** [DigitalCommons/land-explorer-front-end#197]
**Describe the bug**
If a computer loses internet connection while editing then changes are not saved. This is great. But when connection returns the changes are still not saved until the user makes another change. In an ideal world we want the changes to save just because the connection returns. But for now we simply want to switch to read only and let the user know that Editing will be reneabled when connection returns.

**To Reproduce**
1. Save a map
2. Draw something. See it save.
3. Disconnect from internet.
4. Draw something. See error.
5. Reconnect to internet
6. See that nothing is saved until you make a change

**Expected behaviour**
1. Save a map
2. Draw something. See it save.
3. Disconnect from internet.
4. Drawing and editing functionality is disabled.
5. Reconnect to internet
6. Drawing and editing functionality is re enabled

**Context (please complete the following information):**
 - Device: [e.g. iPhone6]: PC
 - OS: [e.g. iOS8.1]: Ubuntu
 - Browser [e.g. stock browser, safari]: Chrome
 - Version [e.g. 22]: Staging 14/4

**Additional context**

This might mean that if connection fails we need to retry saving every X seconds?
~~This will change our autosave flow to save upon change (the subset of changes that trigger an autosave) AND to autoretry if no successful changes saved every X seconds.~~
This might have performance implications? We could also look at doubling the retry time every time save fails. If we do this we'd also need to have a Save Now option.

Drawing and editing = draw functions, editing map name, editing name and description boxes, saving map, sharing map.

Questions:
- Can we still download maps? Or should this be disabled?
- Can we still turn on and off layers? Which ones? Do we need to disabled Property Boundaries layer?

**Comment by rogup** [comment]
@lin-d-hop I suggseted making a separate issue to address connectivity-related behaviour like this: https://github.com/DigitalCommons/land-explorer-front-end/issues/137#issuecomment-1460218536

> Then, in a separate issue, add connectivity-related behaviour. e.g.:
> 
> - add logic to periodically check our connection
> - show a warning and switch to read-only mode when there isn't a connection, then switch off read-only when the connection returns

I think ideally we should go to read-only mode because some operations won't be saved when connection returns, unless we remember them in a queue (e.g. copying objects between maps). Or we should display in the UI when these actions fail (since they just fail silently at the moment).

Are you proposing that we ignore those actions for now, and just periodically retry saving what we can? Your suggestion of retrying saves with exponential backoff sounds good!

**Comment by lin-d-hop** [comment]
Thanks @rogup 

When I was playing I felt like switching to readonly would be frustrating but I also hear you and am aware that the scale of the changes to get to a queue to save on connection returning is bigger than original scope.

I just wrote a really long comment on this trying to explore the alternative. But in writing it I talked myself back out of saving on connection return and I'm back to our original agreement. Let's switch to read only.

I'll update this issue to reflect that. Sorry for the confusion :)
